File: historical/milinda/src/org/openexi/proc/io/BitInputStream.java
package org.openexi.proc.io;

import java.io.IOException;
import java.io.InputStream;

public abstract class BitInputStream extends InputStream {

  InputStream in;
  int getCount;
  int bitBuf;
  
  BitInputStream() {
    getCount = 0;
    bitBuf = 0;
  }
  
  public abstract boolean getBit() throws IOException;

  public abstract int getBits(int n) throws IOException;

}

File: nagasenacs/src/Nagasena/Proc/IO/BitInputStream.cs
using System.IO;

namespace Nagasena.Proc.IO {

  public abstract class BitInputStream {

    internal Stream @in;
    internal int nBits;
    internal int bitBuf;

    internal BitInputStream() {
      nBits = 0;
      bitBuf = 0;
    }

    public abstract bool Bit { get; }

    public abstract int getBits(int n);

    public abstract int OneBit { get; }

    public abstract int TwoBits { get; }

    public abstract int ThreeBits { get; }

    public abstract int FourBits { get; }

    public abstract int FiveBits { get; }

    public abstract int EightBits { get; }

  }

}